src/Form/LegacyConsumer/Actions/DetermineVisibilityForRequest.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Give\Form\LegacyConsumer\Actions;
+
+use ArrayObject;
+use Give\Framework\FieldsAPI\Conditions\BasicCondition;
+use Give\Framework\FieldsAPI\Conditions\Condition;
+use Give\Framework\FieldsAPI\Field;
+use Give\ValueObjects\Money;
+
+/**
+ * @unreleased
+ */
+class DetermineVisibilityForRequest
+{
+    /** @var bool */
+    const IS_VISIBLE = true;
+
+    /** @var Field */
+    private $field;
+
+    /** @var ArrayObject */
+    protected $postData;
+
+    /**
+     * @unreleased
+     * @param Field $field
+     * @param array $postData
+     */
+    public function __construct(Field $field, array $postData )
+    {
+        $this->field = $field;
+        $this->postData = new ArrayObject( $postData );
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        if( ! $this->fieldHasVisibilityConditions() ) {
+            return self::IS_VISIBLE;
+        }
+
+        $conditions = $this->field->getVisibilityConditions();
+        return array_reduce( $conditions, [$this, 'reduceVisibility'], self::IS_VISIBLE );
+    }
+
+    protected function fieldHasVisibilityConditions()
+    {
+        return method_exists( $this->field, 'hasVisibilityConditions' )
+            && $this->field->hasVisibilityConditions();
+    }
+
+    protected function reduceVisibility( $visibility, Condition $condition )
+    {
+        $result = $this->compareConditionWithOperator( $condition );
+
+        return 'and' === $condition->boolean
+            ? $visibility && $result
+            : $visibility || $result;
+    }
+
+    protected function compareConditionWithOperator( Condition $condition )
+    {
+        if( is_a( $condition, BasicCondition::class ) ) {
+            return $this->compareBasicConditionWithOperator( $condition );
+        }
+
+        // @TODO Implement nested conditions.
```

Comment:
I don't think the Fields API supports nested conditions yet, although the groundwork is in place in the API structure.

Reply:
Yeah, nested conditions are not supported yet so we don't have to worry about that now.